src/Evolution/DiscontinuousGalerkin/Messages/BoundaryMessage.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <ostream>
+#include <type_traits>
+
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/PrettyType.hpp"
+
+#include "Evolution/DiscontinuousGalerkin/Messages/BoundaryMessage.decl.h"
+
+namespace evolution::dg {
+/*!
+ * \brief [Charm++ Message]
+ * (https://charm.readthedocs.io/en/latest/charm%2B%2B/manual.html#messages)
+ * intended to be used in `receive_data` calls on the elements to send boundary
+ * data from one element on one node, to a different element on a (potentially)
+ * different node.
+ *
+ * If this message is to be sent across nodes, the `pack()` and `unpack()`
+ * methods will be called on the sending and receiving node, respectively.
+ */
+template <size_t Dim>
+struct BoundaryMessage : public CMessage_BoundaryMessage<Dim> {
+  using base = CMessage_BoundaryMessage<Dim>;
+
+  size_t subcell_ghost_data_size;
+  size_t dg_flux_data_size;
+  bool sent_across_nodes;
```

Comment:
Happy to delay to a followup PR: maybe also send `sender_node` and `sender_core`? That'll be helpful in the future for some optimizations.